### ISSUE TITLE
[StarOS] 'Cannot open null host' error

### ISSUE THREAD

**Issue by tiannan-sr** [StarRocks/starrocks#17928]
### Steps to reproduce the behavior (Required)
````sql
mysql> show backends;
+-----------+---------------+---------------+--------+----------+----------+---------------------+---------------------+-------+----------------------+-----------------------+-----------+------------------+---------------+---------------+---------+----------------+--------+-----------------+--------------------------------------------------------+-------------------+-------------+----------+-------------------+------------+------------+-------------+----------+
| BackendId | IP            | HeartbeatPort | BePort | HttpPort | BrpcPort | LastStartTime       | LastHeartbeat       | Alive | SystemDecommissioned | ClusterDecommissioned | TabletNum | DataUsedCapacity | AvailCapacity | TotalCapacity | UsedPct | MaxDiskUsedPct | ErrMsg | Version         | Status                                                 | DataTotalCapacity | DataUsedPct | CpuCores | NumRunningQueries | MemUsedPct | CpuUsedPct | StarletPort | WorkerId |
+-----------+---------------+---------------+--------+----------+----------+---------------------+---------------------+-------+----------------------+-----------------------+-----------+------------------+---------------+---------------+---------+----------------+--------+-----------------+--------------------------------------------------------+-------------------+-------------+----------+-------------------+------------+------------+-------------+----------+
| 10005     | 172.26.92.200 | 9354          | 9360   | 8338     | 8360     | 2023-02-16 15:43:36 | 2023-02-16 16:09:44 | true  | false                | false                 | 36        | 2.980 MB         | 893.324 GB    | 1.968 TB      | 55.68 % | 55.68 %        |        | UNKNOWN-46580be | {"lastSuccessReportTabletsTime":"2023-02-16 16:09:37"} | 893.327 GB        | 0.00 %      | 16       | 0                 | 2.22 %     | 0.0 %      | 8167        | 1        |
| 10114     | 172.26.92.201 | 9354          | 9360   | 8338     | 8360     | 2023-02-16 15:42:51 | 2023-02-16 16:09:44 | true  | false                | false                 | 41        | 3.980 MB         | 1.283 TB      | 1.968 TB      | 34.81 % | 34.81 %        |        | UNKNOWN-46580be | {"lastSuccessReportTabletsTime":"2023-02-16 16:08:51"} | 1.283 TB          | 0.00 %      | 16       | 0                 | 2.44 %     | 0.0 %      | 8167        | 2        |
| 10115     | 172.26.92.202 | 9354          | 9360   | 8338     | 8360     | 2023-02-16 15:42:56 | 2023-02-16 16:09:44 | true  | false                | false                 | 36        | 2.978 MB         | 1.229 TB      | 1.968 TB      | 37.54 % | 37.54 %        |        | UNKNOWN-46580be | {"lastSuccessReportTabletsTime":"2023-02-16 16:08:57"} | 1.229 TB          | 0.00 %      | 16       | 0                 | 2.30 %     | 0.0 %      | 8167        | 3        |
+-----------+---------------+---------------+--------+----------+----------+---------------------+---------------------+-------+----------------------+-----------------------+-----------+------------------+---------------+---------------+---------+----------------+--------+-----------------+--------------------------------------------------------+-------------------+-------------+----------+-------------------+------------+------------+-------------+----------+
3 rows in set (0.01 sec)

mysql> use tn_test;
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed
mysql> show tables;
+-------------------+
| Tables_in_tn_test |
+-------------------+
| detail_demo       |
+-------------------+
1 row in set (0.00 sec)

mysql> CREATE TABLE `primary_table_without_null` (
        ->    `k1` date NOT NULL,
        ->    `k2` datetime NOT NULL,
        ->    `k3` string NOT NULL,
        ->    `k4` string NOT NULL,
        ->    `k5` boolean NOT NULL,
        ->    `v1` tinyint NOT NULL,
        ->    `v2` smallint NOT NULL,
        ->    `v3` int NOT NULL,
        ->    `v4` bigint NOT NULL,
        ->    `v5` largeint NOT NULL,
        ->    `v6` float NOT NULL,
        ->    `v7` double NOT NULL,
        ->    `v8` decimal(27,9) NOT NULL
        ->  )
        ->  PRIMARY KEY(`k1`, `k2`)
        ->  COMMENT "OLAP"
        ->  DISTRIBUTED BY HASH(`k1`, `k2`) BUCKETS 3
        ->  PROPERTIES (
        ->    "replication_num" = "3",
        ->    "enable_persistent_index" = "true",
        ->    "storage_format" = "v2"
        ->  );
ERROR 1064 (HY000): Cannot open null host.
mysql> show backends;
ERROR 1064 (HY000): Cannot open null host.
mysql> show backends;
````

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
46580be
```

**Comment by sduzh** [comment]
fe.warn.log
```
2023-02-16 16:19:51,303 WARN (starrocks-mysql-nio-pool-45|4018) [StmtExecutor.execute():543] execute Exception, sql show backends
org.apache.thrift.transport.TTransportException: Cannot open null host.
        at org.apache.thrift.transport.TSocket.open(TSocket.java:210) ~[libthrift-0.13.0.jar:0.13.0]
        at com.starrocks.common.GenericPool$ThriftClientFactory.create(GenericPool.java:144) ~[starrocks-fe.jar:?]
        at com.starrocks.common.GenericPool$ThriftClientFactory.create(GenericPool.java:129) ~[starrocks-fe.jar:?]
        at org.apache.commons.pool2.BaseKeyedPooledObjectFactory.makeObject(BaseKeyedPooledObjectFactory.java:62) ~[commons-pool2-2.3.jar:2.3]
        at org.apache.commons.pool2.impl.GenericKeyedObjectPool.create(GenericKeyedObjectPool.java:1036) ~[commons-pool2-2.3.jar:2.3]
        at org.apache.commons.pool2.impl.GenericKeyedObjectPool.borrowObject(GenericKeyedObjectPool.java:356) ~[commons-pool2-2.3.jar:2.3]
        at org.apache.commons.pool2.impl.GenericKeyedObjectPool.borrowObject(GenericKeyedObjectPool.java:278) ~[commons-pool2-2.3.jar:2.3]
        at com.starrocks.common.GenericPool.borrowObject(GenericPool.java:105) ~[starrocks-fe.jar:?]
        at com.starrocks.rpc.FrontendServiceProxy.call(FrontendServiceProxy.java:33) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.LeaderOpExecutor.forward(LeaderOpExecutor.java:169) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.LeaderOpExecutor.execute(LeaderOpExecutor.java:95) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.forwardToLeader(StmtExecutor.java:618) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:413) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:342) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:459) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:723) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:68) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
```

**Comment by sduzh** [comment]
```
MySQL [(none)]> show proc '/frontends';
+----------------------------------+---------------+-------------+----------+-----------+---------+----------+-----------+------+-------+-------------------+---------------------+----------+--------+---------------------+-----------------+
| Name                             | IP            | EditLogPort | HttpPort | QueryPort | RpcPort | Role     | ClusterId | Join | Alive | ReplayedJournalId | LastHeartbeat       | IsHelper | ErrMsg | StartTime           | Version         |
+----------------------------------+---------------+-------------+----------+-----------+---------+----------+-----------+------+-------+-------------------+---------------------+----------+--------+---------------------+-----------------+
| xx.201_9312_1676451442498 | xx.xx.xx.201 | 9312        | 8334     | 9332      | 9322    | FOLLOWER | 252427775 | true | true  | 107869                | 2023-02-16 16:10:44 | true     |        | 2023-02-15 17:20:23 | UNKNOWN-46580be |
| xx.202_9312_1676451775517 | xx.xx.xx.202 | 9312        | 8334     | 9332      | 9322    | FOLLOWER | 252427775 | true | true  | 107869                | 2023-02-16 16:10:44 | true     |        | 2023-02-15 17:21:44 | UNKNOWN-46580be |
| xx.200_9312_1676449450677 | xx.xx.xx.200 | 9312        | 8334     | 9332      | 9322    | FOLLOWER | 252427775 | true | true  | 108778                | 2023-02-16 16:10:44 | true     |        | 2023-02-15 17:18:54 | UNKNOWN-46580be |
+----------------------------------+---------------+-------------+----------+-----------+---------+----------+-----------+------+-------+-------------------+---------------------+----------+--------+---------------------+-----------------+
3 rows in set (0.01 sec)
```

**Comment by sduzh** [comment]
fe.warn.log
```
2023-02-16 16:19:49,943 ERROR (replayer|81) [GlobalStateMgr$5.runOneCycle():1828] replayer thread catch an exception when replay journal 107935.
com.starrocks.journal.JournalException: fail to read journal entity key=107935, data=<DatabaseEntry offset="0" size="5613" data="39 18 0 0 0 3 0 0 0 0 0 1 17 214 0 0 0 0 0 1 17 238 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 0 0 0 0 0 1 18 15 0 0 0 13 112 50 48 50 48 48 54 95 55 48 49 53 56 0 0 0 6 78 79 82 77 65 76 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 0 0 0 0 0 1 17 239 0 0 0  ... 5513 bytes not shown "/>
        at com.starrocks.journal.bdbje.BDBJournalCursor.deserializeData(BDBJournalCursor.java:257) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.bdbje.BDBJournalCursor.next(BDBJournalCursor.java:295) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1954) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.runOneCycle(GlobalStateMgr.java:1819) [starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.run(GlobalStateMgr.java:1884) [starrocks-fe.jar:?]
Caused by: java.io.EOFException
        at java.io.DataInputStream.readFully(DataInputStream.java:197) ~[?:1.8.0_252]
        at com.starrocks.common.io.Text.readString(Text.java:390) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.Replica.readFields(Replica.java:571) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.Replica.read(Replica.java:582) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.LocalTablet.readFields(LocalTablet.java:388) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.LocalTablet.read(LocalTablet.java:404) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.MaterializedIndex.readFields(MaterializedIndex.java:297) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.MaterializedIndex.read(MaterializedIndex.java:308) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.Partition.readFields(Partition.java:419) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.Partition.read(Partition.java:372) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.PartitionPersistInfo.readFields(PartitionPersistInfo.java:133) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.PartitionPersistInfo.read(PartitionPersistInfo.java:126) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.AddPartitionsInfo.read(AddPartitionsInfo.java:49) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.JournalEntity.readFields(JournalEntity.java:278) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.bdbje.BDBJournalCursor.deserializeData(BDBJournalCursor.java:251) ~[starrocks-fe.jar:?]
        ... 5 more
```

**Comment by sduzh** [comment]
> fe.warn.log
> 
> ```
> 2023-02-16 16:19:49,943 ERROR (replayer|81) [GlobalStateMgr$5.runOneCycle():1828] replayer thread catch an exception when replay journal 107935.
> com.starrocks.journal.JournalException: fail to read journal entity key=107935, data=<DatabaseEntry offset="0" size="5613" data="39 18 0 0 0 3 0 0 0 0 0 1 17 214 0 0 0 0 0 1 17 238 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 0 0 0 0 0 1 18 15 0 0 0 13 112 50 48 50 48 48 54 95 55 48 49 53 56 0 0 0 6 78 79 82 77 65 76 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 0 0 0 0 0 1 17 239 0 0 0  ... 5513 bytes not shown "/>
>         at com.starrocks.journal.bdbje.BDBJournalCursor.deserializeData(BDBJournalCursor.java:257) ~[starrocks-fe.jar:?]
>         at com.starrocks.journal.bdbje.BDBJournalCursor.next(BDBJournalCursor.java:295) ~[starrocks-fe.jar:?]
>         at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1954) ~[starrocks-fe.jar:?]
>         at com.starrocks.server.GlobalStateMgr$5.runOneCycle(GlobalStateMgr.java:1819) [starrocks-fe.jar:?]
>         at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
>         at com.starrocks.server.GlobalStateMgr$5.run(GlobalStateMgr.java:1884) [starrocks-fe.jar:?]
> Caused by: java.io.EOFException
>         at java.io.DataInputStream.readFully(DataInputStream.java:197) ~[?:1.8.0_252]
>         at com.starrocks.common.io.Text.readString(Text.java:390) ~[starrocks-fe.jar:?]
>         at com.starrocks.catalog.Replica.readFields(Replica.java:571) ~[starrocks-fe.jar:?]
>         at com.starrocks.catalog.Replica.read(Replica.java:582) ~[starrocks-fe.jar:?]
>         at com.starrocks.catalog.LocalTablet.readFields(LocalTablet.java:388) ~[starrocks-fe.jar:?]
>         at com.starrocks.catalog.LocalTablet.read(LocalTablet.java:404) ~[starrocks-fe.jar:?]
>         at com.starrocks.catalog.MaterializedIndex.readFields(MaterializedIndex.java:297) ~[starrocks-fe.jar:?]
>         at com.starrocks.catalog.MaterializedIndex.read(MaterializedIndex.java:308) ~[starrocks-fe.jar:?]
>         at com.starrocks.catalog.Partition.readFields(Partition.java:419) ~[starrocks-fe.jar:?]
>         at com.starrocks.catalog.Partition.read(Partition.java:372) ~[starrocks-fe.jar:?]
>         at com.starrocks.persist.PartitionPersistInfo.readFields(PartitionPersistInfo.java:133) ~[starrocks-fe.jar:?]
>         at com.starrocks.persist.PartitionPersistInfo.read(PartitionPersistInfo.java:126) ~[starrocks-fe.jar:?]
>         at com.starrocks.persist.AddPartitionsInfo.read(AddPartitionsInfo.java:49) ~[starrocks-fe.jar:?]
>         at com.starrocks.journal.JournalEntity.readFields(JournalEntity.java:278) ~[starrocks-fe.jar:?]
>         at com.starrocks.journal.bdbje.BDBJournalCursor.deserializeData(BDBJournalCursor.java:251) ~[starrocks-fe.jar:?]
>         ... 5 more
> ```

This is the root cause, a broken journal caused the FE node unable to receive the leader switch journal, so the FE node dit not know which FE is leader, i.e, the saved IP address of the leader node is a null.

**Comment by sduzh** [comment]
> > fe.warn.log
> > ```
> > 2023-02-16 16:19:49,943 ERROR (replayer|81) [GlobalStateMgr$5.runOneCycle():1828] replayer thread catch an exception when replay journal 107935.
> > com.starrocks.journal.JournalException: fail to read journal entity key=107935, data=<DatabaseEntry offset="0" size="5613" data="39 18 0 0 0 3 0 0 0 0 0 1 17 214 0 0 0 0 0 1 17 238 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 0 0 0 0 0 1 18 15 0 0 0 13 112 50 48 50 48 48 54 95 55 48 49 53 56 0 0 0 6 78 79 82 77 65 76 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 0 0 0 0 0 1 17 239 0 0 0  ... 5513 bytes not shown "/>
> >         at com.starrocks.journal.bdbje.BDBJournalCursor.deserializeData(BDBJournalCursor.java:257) ~[starrocks-fe.jar:?]
> >         at com.starrocks.journal.bdbje.BDBJournalCursor.next(BDBJournalCursor.java:295) ~[starrocks-fe.jar:?]
> >         at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1954) ~[starrocks-fe.jar:?]
> >         at com.starrocks.server.GlobalStateMgr$5.runOneCycle(GlobalStateMgr.java:1819) [starrocks-fe.jar:?]
> >         at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
> >         at com.starrocks.server.GlobalStateMgr$5.run(GlobalStateMgr.java:1884) [starrocks-fe.jar:?]
> > Caused by: java.io.EOFException
> >         at java.io.DataInputStream.readFully(DataInputStream.java:197) ~[?:1.8.0_252]
> >         at com.starrocks.common.io.Text.readString(Text.java:390) ~[starrocks-fe.jar:?]
> >         at com.starrocks.catalog.Replica.readFields(Replica.java:571) ~[starrocks-fe.jar:?]
> >         at com.starrocks.catalog.Replica.read(Replica.java:582) ~[starrocks-fe.jar:?]
> >         at com.starrocks.catalog.LocalTablet.readFields(LocalTablet.java:388) ~[starrocks-fe.jar:?]
> >         at com.starrocks.catalog.LocalTablet.read(LocalTablet.java:404) ~[starrocks-fe.jar:?]
> >         at com.starrocks.catalog.MaterializedIndex.readFields(MaterializedIndex.java:297) ~[starrocks-fe.jar:?]
> >         at com.starrocks.catalog.MaterializedIndex.read(MaterializedIndex.java:308) ~[starrocks-fe.jar:?]
> >         at com.starrocks.catalog.Partition.readFields(Partition.java:419) ~[starrocks-fe.jar:?]
> >         at com.starrocks.catalog.Partition.read(Partition.java:372) ~[starrocks-fe.jar:?]
> >         at com.starrocks.persist.PartitionPersistInfo.readFields(PartitionPersistInfo.java:133) ~[starrocks-fe.jar:?]
> >         at com.starrocks.persist.PartitionPersistInfo.read(PartitionPersistInfo.java:126) ~[starrocks-fe.jar:?]
> >         at com.starrocks.persist.AddPartitionsInfo.read(AddPartitionsInfo.java:49) ~[starrocks-fe.jar:?]
> >         at com.starrocks.journal.JournalEntity.readFields(JournalEntity.java:278) ~[starrocks-fe.jar:?]
> >         at com.starrocks.journal.bdbje.BDBJournalCursor.deserializeData(BDBJournalCursor.java:251) ~[starrocks-fe.jar:?]
> >         ... 5 more
> > ```
> 
> This is the root cause, a broken journal caused the FE node unable to receive the leader switch journal, so the FE node dit not know which FE is leader, i.e, the saved IP address of the leader node is a null.

Here is one way to reproduce the problem:
```
CREATE TABLE t0(c0 INT) ENGINE=STARROCKS DUPLICATE KEY(c0) DISTRIBUTED BY HASH(c0);
INSERT INTO t0 VALUES(1);
INSERT OVERWRITE t0 VALUES(2);
```

**Comment by sduzh** [comment]
Should have been fixed by #18142, feel free to reopen if meet again.

**Comment by wyb** [comment]
fixed by #17576
